repos/fdbt-netex-output/src/netex-convertor/handler.ts

```diff
@@ -45,7 +48,53 @@ const uploadToS3 = async (netex: string, fileName: string): Promise<void> => {
     }
 };
 
-export const generateFileName = (eventFileName: string): string => eventFileName.replace('.json', '.xml');
+const getLineOrNetworkFare = (productType: string): string => {
+    if (productType === 'singleTrip' || productType === 'dayReturnTrip' || productType === 'periodReturnTrip') {
+        return 'LINE-FARE';
+    }
+
+    return 'NETWORK-FARE';
+};
+
+export const getSchemeNocIdentifier = (schemeTicket: SchemeOperatorTicket): string =>
+    `${schemeTicket.schemeOperatorName.substring(0, 5)}${schemeTicket.schemeOperatorRegionCode}`;
+
+export const getS3FilePrefix = (s3FileName: string, nocIdentifier: string): string => {
+    const exportDate = s3FileName.split(`${nocIdentifier}/exports/`)[1].split('/')[0];
+    return `${nocIdentifier}/exports/${exportDate}`;
+};
+
+export const generateFileName = (ticket: Ticket): string => {
+    const productType = getProductType(ticket);
+    const lineOrNetworkFare = getLineOrNetworkFare(productType);
+    const nocOrSchemeName = isSchemeOperatorTicket(ticket) ? getSchemeNocIdentifier(ticket) : ticket.nocCode;
+    const productName = ticket.products[0].productName.replace(' ', '-');
+    const creationDate = new Date(Date.now()).toISOString().split('T')[0];
+    const startDate = ticket.ticketPeriod.startDate.split('T')[0];
+    return `FX-PI-01_UK_${nocOrSchemeName}_${lineOrNetworkFare}_${productName}_${creationDate}_${startDate}`;
+};
+
+export const checkToSeeIfFileNameExists = async (fileName: string): Promise<string> => {
+    let placeHolderFileName = fileName;
+    let cannotProceed = true;
+    let counter = 1;
+
+    while (cannotProceed) {
+        const fileNameTaken = await fileNameExistsAlready(placeHolderFileName);
+        if (fileNameTaken) {
+            if (counter > 1) {
+                placeHolderFileName = `${placeHolderFileName.substring(0, placeHolderFileName.length - 2)}_${counter}`;
```

Comment:
can we do it like this:

placeHolderFileName = `${fileName}_${counter}`;

Reply:
The problem if we do that, is that if it fails the check again, we will end up with a name like `<fileName>_1_2`